src/main/java/racingcar/domain/RacingCarGame.java

```diff
@@ -6,51 +6,71 @@
 import java.util.stream.Collectors;
-import java.util.stream.IntStream;
+import java.util.stream.StreamSupport;
 
 public class RacingCarGame {
-    private final List<Car> carList;
+    private static final int FIRST_INDEX = 0;
 
-    public RacingCarGame(List<Car> carList) {
-        this.carList = carList;
+    private final StrategyRandomMove strategyRandomMove = new StrategyRandomMove();
+    private final Map<Participant, Car> racingCars;
+
+    public RacingCarGame(Map<Participant, Car> racingCars) {
+        this.racingCars = racingCars;
     }
 
-    public static RacingCarGame from(int carCount) {
-        return new RacingCarGame(createCarList(carCount));
+    public static RacingCarGame from(Participants participants) {
+        return new RacingCarGame(createRacingCars(participants));
     }
-    public static List<Car> createCarList(int carCount) {
-        return IntStream.range(0, carCount)
-                .mapToObj(i -> new Car())
-                .collect(Collectors.toList());
+    public static Map<Participant, Car> createRacingCars(Participants participants) {
+        return StreamSupport.stream(participants.spliterator(), false)
+                .collect(Collectors.toMap(
+                        participant -> participant,
+                        participant -> new Car(),
+                        (existing, replacement) -> existing,
+                        LinkedHashMap::new
+                ));
     }
 
     public RacingResult executeRacing(int trialCount) {
         List<RoundResult> roundResults = new ArrayList<>();
         while (trialCount-- > 0) {
             roundResults.add(executeRound());
         }
-        return new RacingResult(roundResults);
+        return new RacingResult(roundResults, selectWinners());
     }
 
     public RoundResult executeRound() {
-        play(new StrategyRandomMove());
+        play(strategyRandomMove);
         return new RoundResult(getRoundResults());
     }
 
     private List<RoundScore> getRoundResults() {
-        List<RoundScore> roundScores = new ArrayList<>();
-        for (Car car : carList) {
-            roundScores.add(new RoundScore(car.getPosition()));
-        }
-        return roundScores;
+        return racingCars.keySet().stream()
+                .map(participant -> new RoundScore(participant.getName(), racingCars.get(participant).getPosition()))
+                .collect(Collectors.toList());
     }
 
     private void play(MoveStrategy moveStrategy) {
-        for (Car car : carList) {
-            car.moveForward(moveStrategy);
-        }
+        racingCars.keySet().forEach(participant -> racingCars.get(participant).moveForward(moveStrategy));
+    }
+
+    private Participants selectWinners() {
+        int winnerPosition = calculateWinnerPosition();
+        return new Participants(
+                racingCars.keySet().stream()
+                        .filter(participant -> racingCars.get(participant).getPosition() == winnerPosition)
```

Comment:
OOP의 특징 중 하나는 객체간의 요청과 응답입니다. 
그런데 이처럼 객체의 내부 값을 직접 꺼내 연산을 수행해버리면 사실 내부의 값을 private로 접근제어를 할 필요가 없어보이죠. 객체에 요청을 하는식으로 바꿔볼 수 있지 않을까요?

```suggestion
                        .filter(participant -> racingCars.get(participant).isEqualsPosition(winnerPosition))
```